server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -364,7 +364,7 @@ public ClusterState execute(ClusterState currentState) {
                     repositoryData.resolveIndices(matchingIndices),
                     threadPool.absoluteTimeInMillis(),
                     repositoryData.getGenId(),
-                    minCompatibleVersion(currentState.nodes().getMinNodeVersion(), repositoryData, null)
+                    minCompatibleVersion(currentState.nodes().getMaxDataNodeCompatibleIndexVersion(), repositoryData, null)
```

Comment:
These are not identical - `getMinNodeVersion` considers all nodes, `getMaxDataNodeCompatibleIndexVersion` only considers data/master nodes. But snapshots are only relevant to data nodes, AFAIK.

Reply:
Jup this should be ok, only master and data nodes interact with the repository.